tests/scripts/test_config_script.py

```diff
@@ -147,10 +147,12 @@ def run_all(options):
     for preset in presets:
         run_one(options, [preset])
     for symbol in TEST_SYMBOLS:
+        run_one(options, ['get', symbol])
         run_one(options, ['set', symbol])
         run_one(options, ['--force', 'set', symbol])
         run_one(options, ['unset', symbol])
     for symbol in TEST_SYMBOLS_WITH_VALUE:
+        run_one(options, ['get', symbol])
```

Comment:
This script doesn't check anything for sanity. It's purely a non-regression test: record the outcome, and then you can compare it with the previous version. See the comment at the top of the script and the PR description.

Reply:
How do you know that get works as well as it worked with config.pl?

Reply:
I run
```
./tests/scripts/test_config_script.py -d old -s ./config.pl 
./tests/scripts/test_config_script.py -d new
diff -ruw old new
```
and it doesn't find any difference.

Reply:
When I run `./tests/scripts/test_config_script.py -d old -s ./config.pl`, I get an error about `./config.pl` not found. When I run `./tests/scripts/test_config_script.py -d old -s ./scripts/config.pl`, I see no output at all.

Reply:
You need to obtain the old `config.pl` for comparison. See the PR description.

Seeing no output is expected. The data is written to the directory specified with `-d`. Don't run `test_config_script.py -d old -s ./scripts/config.pl` on the PR: that would test the new wrapper. But you can run that on a checkout of `development`.

Reply:
I had to run `git show development:scripts/config.pl >|config.pl`. Missed that step.

Reply:
After running the test, I see the .out file is empty for the value parameters (probably because the value isn’t set in the default config.h to any value). We could use `configs/config-suite-b.h` to get a value for `MBEDTLS_MPI_MAX_SIZE`. `CUSTOM_VALUE` isn't set to any value in any of our configs.

Reply:
I've added tests of `set FOO value` followed by `get`. They pass.

Reply:
I see this diff now (on my Mac). I'll try Linux next.

```
> diff -ruw old new
diff -ruw old/config-baremetal.h new/config-baremetal.h
--- old/config-baremetal.h	2019-09-19 13:47:36.000000000 +0100
+++ new/config-baremetal.h	2019-09-19 13:47:39.000000000 +0100
@@ -729,7 +729,7 @@
  *
  * Comment this macro to keep 3DES in the default ciphersuite list.
  */
-#define MBEDTLS_REMOVE_3DES_CIPHERSUITES
+//#define MBEDTLS_REMOVE_3DES_CIPHERSUITES
 
 /**
  * \def MBEDTLS_ECP_DP_SECP192R1_ENABLED
@@ -787,7 +787,7 @@
  *        MBEDTLS_ECP_ALT, MBEDTLS_ECDH_XXX_ALT, MBEDTLS_ECDSA_XXX_ALT
  *        and MBEDTLS_ECDH_LEGACY_CONTEXT.
  */
-#define MBEDTLS_ECP_RESTARTABLE
+//#define MBEDTLS_ECP_RESTARTABLE
 
 /**
  * \def MBEDTLS_ECDH_LEGACY_CONTEXT
@@ -1230,7 +1230,7 @@
  *
  * Uncomment this macro to let the buffer allocator print out error messages.
  */
-#define MBEDTLS_MEMORY_DEBUG
+//#define MBEDTLS_MEMORY_DEBUG
 
 /**
  * \def MBEDTLS_MEMORY_BACKTRACE
@@ -1287,7 +1287,7 @@
  * Requires: MBEDTLS_PSA_CRYPTO_C
  *
  */
-#define MBEDTLS_PSA_CRYPTO_SPM
+//#define MBEDTLS_PSA_CRYPTO_SPM
 
 /**
  * \def MBEDTLS_PSA_INJECT_ENTROPY
@@ -1300,7 +1300,7 @@
  * Requires: MBEDTLS_PSA_CRYPTO_STORAGE_C, MBEDTLS_ENTROPY_NV_SEED
  *
  */
-#define MBEDTLS_PSA_INJECT_ENTROPY
+//#define MBEDTLS_PSA_INJECT_ENTROPY
 
 /**
  * \def MBEDTLS_RSA_NO_CRT
@@ -2897,7 +2897,7 @@
  *           either MBEDTLS_PSA_ITS_FILE_C or a native implementation of
  *           the PSA ITS interface
  */
-#define MBEDTLS_PSA_CRYPTO_STORAGE_C
+//#define MBEDTLS_PSA_CRYPTO_STORAGE_C
 
 /**
  * \def MBEDTLS_PSA_ITS_FILE_C
@@ -2910,7 +2910,7 @@
  * Requires: MBEDTLS_FS_IO
  *
  */
-#define MBEDTLS_PSA_ITS_FILE_C
+//#define MBEDTLS_PSA_ITS_FILE_C
 
 /**
  * \def MBEDTLS_RIPEMD160_C
diff -ruw old/config-full.h new/config-full.h
--- old/config-full.h	2019-09-19 13:47:36.000000000 +0100
+++ new/config-full.h	2019-09-19 13:47:39.000000000 +0100
@@ -729,7 +729,7 @@
  *
  * Comment this macro to keep 3DES in the default ciphersuite list.
  */
-#define MBEDTLS_REMOVE_3DES_CIPHERSUITES
+//#define MBEDTLS_REMOVE_3DES_CIPHERSUITES
 
 /**
  * \def MBEDTLS_ECP_DP_SECP192R1_ENABLED
@@ -787,7 +787,7 @@
  *        MBEDTLS_ECP_ALT, MBEDTLS_ECDH_XXX_ALT, MBEDTLS_ECDSA_XXX_ALT
  *        and MBEDTLS_ECDH_LEGACY_CONTEXT.
  */
-#define MBEDTLS_ECP_RESTARTABLE
+//#define MBEDTLS_ECP_RESTARTABLE
 
 /**
  * \def MBEDTLS_ECDH_LEGACY_CONTEXT
@@ -1230,7 +1230,7 @@
  *
  * Uncomment this macro to let the buffer allocator print out error messages.
  */
-#define MBEDTLS_MEMORY_DEBUG
+//#define MBEDTLS_MEMORY_DEBUG
 
 /**
  * \def MBEDTLS_MEMORY_BACKTRACE
@@ -1242,7 +1242,7 @@
  *
  * Uncomment this macro to include backtrace information
  */
-#define MBEDTLS_MEMORY_BACKTRACE
+//#define MBEDTLS_MEMORY_BACKTRACE
 
 /**
  * \def MBEDTLS_PK_RSA_ALT_SUPPORT
@@ -1287,7 +1287,7 @@
  * Requires: MBEDTLS_PSA_CRYPTO_C
  *
  */
-#define MBEDTLS_PSA_CRYPTO_SPM
+//#define MBEDTLS_PSA_CRYPTO_SPM
 
 /**
  * \def MBEDTLS_PSA_INJECT_ENTROPY
@@ -1300,7 +1300,7 @@
  * Requires: MBEDTLS_PSA_CRYPTO_STORAGE_C, MBEDTLS_ENTROPY_NV_SEED
  *
  */
-#define MBEDTLS_PSA_INJECT_ENTROPY
+//#define MBEDTLS_PSA_INJECT_ENTROPY
 
 /**
  * \def MBEDTLS_RSA_NO_CRT
@@ -2659,7 +2659,7 @@
  *
  * Enable this module to enable the buffer memory allocator.
  */
-#define MBEDTLS_MEMORY_BUFFER_ALLOC_C
+//#define MBEDTLS_MEMORY_BUFFER_ALLOC_C
 
 /**
  * \def MBEDTLS_NET_C
diff -ruw old/config-set-CUSTOM_SYMBOL-value.status new/config-set-CUSTOM_SYMBOL-value.status
--- old/config-set-CUSTOM_SYMBOL-value.status	2019-09-19 13:47:36.000000000 +0100
+++ new/config-set-CUSTOM_SYMBOL-value.status	2019-09-19 13:47:39.000000000 +0100
@@ -1 +1 @@
-25
+1
diff -ruw old/config-set-CUSTOM_SYMBOL.status new/config-set-CUSTOM_SYMBOL.status
--- old/config-set-CUSTOM_SYMBOL.status	2019-09-19 13:47:36.000000000 +0100
+++ new/config-set-CUSTOM_SYMBOL.status	2019-09-19 13:47:39.000000000 +0100
@@ -1 +1 @@
-25
+1
```

Reply:
Ran again with newest `config.pl` (I had been using an older one).

```
diff -ruw old/config-set-CUSTOM_SYMBOL-value.status new/config-set-CUSTOM_SYMBOL-value.status
--- old/config-set-CUSTOM_SYMBOL-value.status	2019-09-19 13:52:16.000000000 +0100
+++ new/config-set-CUSTOM_SYMBOL-value.status	2019-09-19 13:52:28.000000000 +0100
@@ -1 +1 @@
-25
+1
diff -ruw old/config-set-CUSTOM_SYMBOL.status new/config-set-CUSTOM_SYMBOL.status
--- old/config-set-CUSTOM_SYMBOL.status	2019-09-19 13:52:16.000000000 +0100
+++ new/config-set-CUSTOM_SYMBOL.status	2019-09-19 13:52:28.000000000 +0100
@@ -1 +1 @@
-25
+1
```

Reply:
I still see this diff, even on Linux.

Reply:
```$ ./scripts/config.pl set CUSTOM_SYMBOL value
A #define for the symbol CUSTOM_SYMBOL was not found in include/mbedtls/config.h
$ echo $status
1
$ ./config.pl set CUSTOM_SYMBOL value
A #define for the symbol CUSTOM_SYMBOL was not found in include/mbedtls/config.h
$ echo $status
25
```

Error code returned is different. Not sure we care, but is this different from what you see? You said you see no difference in behavior.

Reply:
OK. I had trouble reconciling what I saw with your comment in "https://github.com/ARMmbed/mbedtls/pull/2765#discussion_r324288055" and wanted to confirm.